### PR TITLE
fix: (Day) Allow bags to move stackable items properly

### PR DESCRIPTION
**Pull request by AlexVild**

### CODE REVIEW

**Review by Arufonsu — APPROVED** [review]
👍🏽 

https://user-images.githubusercontent.com/17498701/174110986-b4c9bdb8-83ff-48e8-8b6e-6cd3a7ca502a.mp4

**Comment by dudevelopement** [comment]
Tested and working perfectly.

File: Intersect.Server/Entities/Player.cs [hunk 1]
```diff
@@ -3898,124 +3908,199 @@ public void CloseBag()
...
+            var bagSlots = bag.Slots;
+            foreach (var inventorySlotsWithItem in inventorySlots)
             {
-                return;
-            }
+                // If we've fulfilled our stacking desires, we're done
+                if (amountRemainder <= 0 || FindOpenInventorySlots().Count <= 0)
+                {
+                    return amountRemainder <= 0;
+                }
```
Comment (lodicolo) [inline Intersect.Server/Entities/Player.cs:3930]: Can you just change this to < 1?

File: Intersect.Server/Entities/Player.cs [hunk 2]
```diff
@@ -3898,124 +3908,199 @@ public void CloseBag()
...
+                if (amountToGive > 0)
+                {
+                    PacketSender.SendInventoryItemUpdate(this, currSlot);
+                    PacketSender.SendBagUpdate(this, bagSlotIdx, bagSlots[bagSlotIdx]);
+                }
+            } // repeat until we've either filled the bag or fulfilled our stack requirements
 
-                        return;
-                    }
+            return amountRemainder <= 0;
```
Comment (lodicolo) [inline Intersect.Server/Entities/Player.cs:3967]: Also `< 1`

File: Intersect.Server/Entities/Player.cs [hunk 3]
```diff
@@ -3898,124 +3908,199 @@ public void CloseBag()
...
+        /// <returns>A <see cref="bool"/> saying whether or not we successfully deposited ALL the items</returns>
+        private bool TryFillBagStacksOfItemFromInventorySlot(Bag bag, int inventorySlotIdx, List<BagSlot> bagSlots, ItemBase itemDescriptor, int amountToGive = 1)
+        {
+            int amountRemainder = amountToGive;
+            foreach (var bagSlotWithItem in bagSlots)
+            {
+                if (amountRemainder <= 0 || bag.FindOpenBagSlots().Count <= 0)
+                {
+                    return amountRemainder <= 0;
+                }
```
Comment (lodicolo) [inline Intersect.Server/Entities/Player.cs:3987]: Same `< 1` stuff

File: Intersect.Server/Entities/Player.cs [hunk 4]
```diff
@@ -3898,124 +3908,199 @@ public void CloseBag()
...
-                                if (amount >= Items[slot].Quantity)
-                                {
-                                    Items[slot].Set(Item.None);
-                                    EquipmentProcessItemLoss(slot);
-                                }
-                                else
-                                {
-                                    Items[slot].Quantity -= amount;
-                                }
+            return amountRemainder <= 0;
```
Comment (lodicolo) [inline Intersect.Server/Entities/Player.cs:4019]: And here

**Review by lodicolo — DISMISSED** [review]

File: Intersect.Server/Entities/Player.cs (hunk 4 above)
Comment (AlexVild) [reply Intersect.Server/Entities/Player.cs:4019]: Done for all

**Review by lodicolo — APPROVED** [review]

**Review by Arufonsu — APPROVED** [review]

**Pull request merged**